packages/x-charts/src/context/SvgRefProvider/SvgRefProvider.tsx

```diff
@@ -1,12 +1,18 @@
 'use client';
 import * as React from 'react';
+import useForkRef from '@mui/utils/useForkRef';
 import { SvgRefProviderProps } from './SvgRef.types';
 import { SvgRefContext } from './SvgRefContext';
 
 export function SvgRefProvider(props: SvgRefProviderProps) {
-  const { svgRef, children } = props;
+  const { svgRef: inRef, children } = props;
+  const svgRef = React.useRef<SVGSVGElement>(null);
+  const surfaceRef = useForkRef(inRef, svgRef);
 
```

Comment:
In this PR the `ChartsSurface` is still inside the data provider 🙃

I'll remove that in https://github.com/mui/mui-x/pull/15375 so the user can only set the refs in the ways you mentioned above. This is just so that this PR doesn't get bigger than the other one while still keeping a releasable state

Reply:
If that's temporary ok :+1: